feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/source/BaseNftTransfer.kt

```diff
@@ -0,0 +1,75 @@
+import io.novafoundation.nova.feature_nft_impl.data.mappers.mapNftTypeLocalToTypeKey
+import io.novafoundation.nova.feature_nft_impl.data.network.blockchain.nfts.transfers.NftTransferModel
+import io.novafoundation.nova.feature_nft_impl.data.network.blockchain.nfts.transfers.NftTransfersValidationSystem
+import io.novafoundation.nova.feature_nft_impl.data.network.blockchain.nfts.transfers.validations.nftExists
+import io.novafoundation.nova.feature_nft_impl.data.network.blockchain.nfts.transfers.validations.notPhishingRecipient
+import io.novafoundation.nova.feature_nft_impl.data.network.blockchain.nfts.transfers.validations.sufficientCommissionBalanceToStayAboveED
+import io.novafoundation.nova.feature_nft_impl.data.network.blockchain.nfts.transfers.validations.sufficientTransferableBalanceToPayOriginFee
+import io.novafoundation.nova.feature_nft_impl.data.network.blockchain.nfts.transfers.validations.validAddress
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.AssetSourceRegistry
+import io.novafoundation.nova.feature_wallet_api.domain.validation.PhishingValidationFactory
+import jp.co.soramitsu.fearless_utils.runtime.extrinsic.ExtrinsicBuilder
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.withContext
+import java.math.BigInteger
+
+abstract class BaseNftTransfer(
+    private val assetSourceRegistry: AssetSourceRegistry,
+    private val extrinsicService: ExtrinsicService,
+    private val phishingValidationFactory: PhishingValidationFactory,
+    private val nftProvidersRegistry: NftProvidersRegistry,
+    private val nftDao: NftDao,
+    private val exceptionHandler: HttpExceptionHandler
+) : NftTransfer {
+
+    abstract fun ExtrinsicBuilder.transfer(transfer: NftTransferModel)
+
+    suspend fun performTransfer(transfer: NftTransferModel): Result<String> {
+        return withContext(Dispatchers.IO) {
+            val senderAccountId = transfer.sender.requireAccountIdIn(transfer.originChain)
+
+            extrinsicService.submitExtrinsicWithAnySuitableWallet(transfer.originChain, senderAccountId) {
+                transfer(transfer)
+            }
+        }
+    }
+
+    suspend fun calculateFee(transfer: NftTransferModel): BigInteger {
+        return withContext(Dispatchers.IO) {
+            extrinsicService.estimateFee(transfer.originChain) {
+                transfer(transfer)
+            }
+        }
+    }
+
+    suspend fun nftDetails(nftId: String): NftDetails {
+        return runCatching {
+            val nftTypeKey = mapNftTypeLocalToTypeKey(nftDao.getNftType(nftId))
+            val nftProvider = nftProvidersRegistry.get(nftTypeKey)
+            nftProvider.nftDetailsFlow(nftId)
+        }.onFailure { throw exceptionHandler.transformException(it) }
+            .getOrThrow()
+            .first()
+    }
+
+    fun defaultValidationSystem(): NftTransfersValidationSystem = ValidationSystem {
```

Comment:
I suggest making those methods available in `NftTransfer` so we can inject `NftTransfer` and not `BaseNftTransfer` which is just base implementation

Reply:
Done